### PR TITLE
Un-hardcode AWS Region

### PR DESCRIPTION
**Pull request by danbills**
Use the config option where appropriate

### CODE REVIEW

**Comment by wleepang** [comment]
Agree with @elerch .

Also, should occurances of `Region.AP_NORTHEAST_1` actually be `Region.US_EAST_1` for this backend?

**Comment by danbills** [comment]
@elerch 
1) changed to `region`
2) agree w/ approach. I'm going to make the config be an Option[Region], then if none will omit the `region()` builder call

**Comment by elerch** [comment]
@danbills - changes lgtm!

File: cloudSupport/src/main/scala/cromwell/cloudsupport/aws/auth/AwsAuthMode.scala [hunk 1]
```diff
@@ -71,18 +71,17 @@ sealed trait AwsAuthMode {
...
-       STSClient
-          .builder
-          .region(Region.of(region))
-          .credentialsProvider(StaticCredentialsProvider.create(credentials))
-          .build
-          .getCallerIdentity(GetCallerIdentityRequest.builder.build)
+   private[auth] var credentialValidation: ((AwsCredentials, Option[String]) => Unit) =
+     (credentials: AwsCredentials, region: Option[String]) => {
+       val builder = STSClient.builder
+       region.foreach(str => builder.region(Region.of(str)))
```
Comment (mcovarr) [inline cloudSupport/src/main/scala/cromwell/cloudsupport/aws/auth/AwsAuthMode.scala]: I have to admit it took me a bit to grok that this is conditionally setting the region into the statefully built builder... maybe a comment?

File: filesystems/s3/src/test/scala/cromwell/filesystems/s3/S3PathBuilderSpec.scala [hunk 2]
```diff
@@ -49,7 +50,8 @@ class S3PathBuilderSpec extends TestKitSuite with FlatSpecLike with Matchers wit
     S3PathBuilder.fromCredentials(
       AnonymousCredentialsProvider.create.getCredentials,
       S3Storage.s3AdvancedConfiguration(),
-      wfOptionsWithProject
+      wfOptionsWithProject,
+      Some(Region.US_EAST_1)
```
Comment (mcovarr) [inline filesystems/s3/src/test/scala/cromwell/filesystems/s3/S3PathBuilderSpec.scala]: `Option` here and below

File: supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala [hunk 3]
```diff
@@ -62,33 +62,36 @@ import scala.util.Try
...
-    //TODO: setting this to get past batch job unit test, should be configured externally
-    .region(Region.US_EAST_1)
-                   // .credentialsProvider(...)
-                   // .endpointOverride(...)
-                   .build
+  lazy val client = {
+   val builder = BatchClient.builder()
+   configRegion.foreach(builder.region)
+      // .credentialsProvider(...)
+      // .endpointOverride(...)
```
Comment (mcovarr) [inline supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala]: ✖️

File: supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala [hunk 4]
```diff
@@ -62,33 +62,36 @@ import scala.util.Try
...
+   builder.build
+  }
+  lazy val logsclient = {
+    val builder = CloudWatchLogsClient.builder()
+    configRegion.foreach(builder.region)
+      // .credentialsProvider(...)
+      // .endpointOverride(...)
```
Comment (mcovarr) [inline supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala]: ✖️

**Pull request merged**